lib/tlIO/STBRead.cpp

```diff
@@ -0,0 +1,151 @@
+// SPDX-License-Identifier: BSD-3-Clause
+// Copyright (c) 2021-2023 Darby Johnston
+// All rights reserved.
+
+#define STBI_NO_JPEG
+#define STBI_NO_PNG
+#define STBI_NO_HDR
+#define STBI_NO_PNM
+#define STB_IMAGE_IMPLEMENTATION
+#include <tlIO/stb_image.h>
+
+#include <tlIO/STB.h>
+
+#include <tlCore/StringFormat.h>
+
+namespace tl
+{
+    namespace stb
+    {
+        namespace
+        {
+            
+            class File
+            {
+            public:
+                File(const std::string& fileName, const file::MemoryRead* memory)
+                {
+                    int w = 0, h = 0, n = 0, bits = 8;
+                    int res = stbi_info(fileName.c_str(), &w, &h, &n);
```

Comment:
Does STB have the ability to read from memory and files?

Reply:
Yes.  I added new code to handle it now.